os/drivers/testcase/test_group.c

```diff
@@ -0,0 +1,262 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <debug.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sched.h>
+#include <sys/types.h>
+#include <tinyara/testcase_drv.h>
+#include <tinyara/sched.h>
+#ifdef CONFIG_SCHED_HAVE_PARENT
+#include <group/group.h>
+#ifdef CONFIG_SCHED_CHILD_STATUS
+#include <task/task.h>
+#endif
+#endif
+#include "kernel_test_proto.h"
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+#if defined(CONFIG_SCHED_HAVE_PARENT) && defined(CONFIG_SCHED_CHILD_STATUS)
+#define TASK_STACKSIZE 2048
+static struct tcb_s *st_tcb;
+static struct task_group_s *group;
+static struct child_status_s *child;
+static struct child_status_s *child_returned;
+static pid_t child_pid;
+#endif
+
+/****************************************************************************
+ * Private Function
+ ****************************************************************************/
+
+#if defined(CONFIG_SCHED_HAVE_PARENT) && defined(CONFIG_SCHED_CHILD_STATUS)
+static int get_tcb(void)
```

Comment:
get_tcb is unnecessary, not important in group testcase. Let's make it as get_group.

Reply:
modify